File: android/src/main/kotlin/com/example/mango_player/ijkplayer/IJKPlayerManager.kt
package com.example.mango_player.ijkplayer

import android.content.Context
import android.view.Surface
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodChannel
import com.example.mango_player.platform.TextureRegistryHandler

class IJKPlayerManager(private val context: Context) {
    private val playerWrapper = IJKPlayerWrapper(context)
    private var eventSink: EventChannel.EventSink? = null
    private var textureRegistryHandler: TextureRegistryHandler? = null
    
    init {
        playerWrapper.initialize()
        setupListeners()
    }
    
    fun setTextureRegistryHandler(handler: TextureRegistryHandler) {
        this.textureRegistryHandler = handler
    }

    private fun setupListeners() {
        val mediaPlayer = playerWrapper.getMediaPlayer()
        mediaPlayer?.setOnPreparedListener { 
            eventSink?.success(mapOf("type" to "state", "state" to "ready"))
            eventSink?.success(mapOf("type" to "progress", "duration" to playerWrapper.getDuration()))
        }
        mediaPlayer?.setOnCompletionListener {
            eventSink?.success(mapOf("type" to "state", "state" to "completed"))
            eventSink?.success(mapOf("type" to "completed"))
        }
        mediaPlayer?.setOnErrorListener { _, what, extra ->
            eventSink?.success(mapOf("type" to "error", "code" to "unknown", "message" to "IJK Error: $what, $extra"))
            true
        }
        mediaPlayer?.setOnBufferingUpdateListener { _, percent ->
             eventSink?.success(mapOf("type" to "buffering", "bufferPercentage" to percent / 100.0))
        }
    }

    fun initialize(uri: String, type: String, headers: Map<String, String>?, result: MethodChannel.Result, textureId: Long?) {
        try {
            if (textureId != null && textureRegistryHandler != null) {
                val entry = textureRegistryHandler!!.getTextureEntry(textureId)
                if (entry != null) {
                    val surface = Surface(entry.surfaceTexture())
                    playerWrapper.setSurface(surface)
                }
            }
            
            playerWrapper.setDataSource(uri, headers)
            playerWrapper.prepareAsync()
            eventSink?.success(mapOf("type" to "state", "state" to "initializing"))
            result.success(mapOf("success" to true, "duration" to 0))
        } catch (e: Exception) {
            result.error("initialization_error", e.message, null)
        }
    }

    fun play() {
        playerWrapper.start()
        eventSink?.success(mapOf("type" to "state", "state" to "playing"))
    }

    fun pause() {
        playerWrapper.pause()
        eventSink?.success(mapOf("type" to "state", "state" to "paused"))
    }

    fun stop() {
        playerWrapper.stop()
        eventSink?.success(mapOf("type" to "state", "state" to "idle"))
    }

    fun seekTo(position: Long) {
        playerWrapper.seekTo(position)
    }

    fun setVolume(volume: Float) {
        playerWrapper.setVolume(volume)
    }

    fun setPlaybackSpeed(speed: Float) {
        playerWrapper.setSpeed(speed)
    }

    fun getPosition(): Long {
        return playerWrapper.getCurrentPosition()
    }

    fun getDuration(): Long {
        return playerWrapper.getDuration()
    }

    fun release() {
        playerWrapper.release()
    }

    fun setEventSink(sink: EventChannel.EventSink?) {
        this.eventSink = sink
    }
}

File: ios/Classes/IJKPlayer/IJKPlayerManager.swift
import Foundation
import Flutter
import IJKMediaFramework

class IJKPlayerManager: NSObject, IJKPlayerFrameDelegate {
    private let playerWrapper = IJKPlayerWrapper()
    private var eventSink: FlutterEventSink?
    private var textureRegistryHandler: TextureRegistryHandler?
    private var currentTextureId: Int64?
    
    override init() {
        super.init()
        playerWrapper.frameDelegate = self
        setupListeners()
    }
    
    func setTextureRegistryHandler(handler: TextureRegistryHandler?) {
        self.textureRegistryHandler = handler
    }
    
    private func setupListeners() {
        NotificationCenter.default.addObserver(self, selector: #selector(onStateChanged), name: NSNotification.Name.IJKMPMoviePlayerLoadStateDidChange, object: nil)
        NotificationCenter.default.addObserver(self, selector: #selector(onPlaybackFinished), name: NSNotification.Name.IJKMPMoviePlayerPlaybackDidFinish, object: nil)
        // Add more listeners
    }
    
    @objc private func onStateChanged(notification: Notification) {
        guard let player = playerWrapper.getPlayer() else { return }
        let loadState = player.loadState
        if loadState.contains(.playthroughOK) {
            eventSink?(["type": "state", "state": "ready"])
            eventSink?(["type": "progress", "duration": playerWrapper.getDuration()])
        } else if loadState.contains(.stalled) {
            eventSink?(["type": "buffering", "isBuffering": true])
        }
    }
    
    @objc private func onPlaybackFinished(notification: Notification) {
        eventSink?(["type": "state", "state": "completed"])
        eventSink?(["type": "completed"])
    }
    
    func initialize(uri: String, type: String, headers: [String: String]?, result: @escaping FlutterResult, textureId: Int64?) {
        self.currentTextureId = textureId
        playerWrapper.createPlayer(uri: uri, headers: headers)
        eventSink?(["type": "state", "state": "initializing"])
        result(["success": true, "duration": 0])
    }
    
    func play() {
        playerWrapper.play()
        eventSink?(["type": "state", "state": "playing"])
    }
    
    func pause() {
        playerWrapper.pause()
        eventSink?(["type": "state", "state": "paused"])
    }
    
    func stop() {
        playerWrapper.stop()
        eventSink?(["type": "state", "state": "idle"])
    }
    
    func seekTo(position: Int64) {
        playerWrapper.seekTo(position: position)
    }
    
    func setVolume(volume: Float) {
        playerWrapper.setVolume(volume: volume)
    }
    
    func setPlaybackSpeed(speed: Float) {
        playerWrapper.setPlaybackSpeed(speed: speed)
    }
    
    func getPosition() -> Int64 {
        return playerWrapper.getPosition()
    }
    
    func getDuration() -> Int64 {
        return playerWrapper.getDuration()
    }
    
    func release() {
        playerWrapper.release()
        NotificationCenter.default.removeObserver(self)
    }
    
    func setEventSink(sink: FlutterEventSink?) {
        self.eventSink = sink
    }
    
    // IJKPlayerFrameDelegate
    func onVideoFrame(pixelBuffer: CVPixelBuffer) {
        if let textureId = currentTextureId, let manager = textureRegistryHandler?.getTextureManager(textureId: textureId) {
            manager.updatePixelBuffer(pixelBuffer)
        }
    }
}
